Documentation/design-docs/finally-optimizations.md

```diff
@@ -175,6 +175,41 @@ G_M60484_IG06:
 Empty finally removal is unconditionally profitable: it should always
 reduce code size and improve code speed.
 
+Empty Try Removal
+---------------------
+
+If the try region of a try-finally is empty, and the jitted code will
+execute on a runtime that does not protect finally execution from
+thread abort, then the try-finally can be replaced with just the
+content of the finally.
+
+Empty trys with non-empty finallys often exist in code that must run
+under both thread-abort aware and non-thread-abort aware runtimes. In
+the former case the placement of cleanup code in thea finally ensures
+that the cleanup code will execute fully. But if thread abort is not
+possible, the extra protection offered by the finally is not needed.
+
+Empty try removal looks for try-finallys where the try region does
+nothing except invoke the finally. There are currently two different
+EH implementation models, so the try screening has two cases:
+
+* callfinally thunks (all but x86): the try must be a single empty
+basic block that always jumps to a callfinally that is the first
+half of a callalways pair;
```

Comment:
>callalways [](start = 10, length = 10)

can you write out "callallways pair" as "callfinally/always pair"? it's easier to see it's actually a "pair" in that case, and it matches the JIT BBJ_* names more closely.